adam_core/backend/pyoorb.py

```diff
@@ -0,0 +1,326 @@
+try:
+    import pyoorb as oo
+except ImportError:
+    raise ImportError("PYOORB is not installed.")
+
+import os
+import warnings
+from typing import Optional
+
+import numpy as np
+from astropy.time import Time
+
+from ..coordinates.cartesian import CartesianCoordinates
+from ..orbits.orbits import Orbits
+from .backend import Backend
+
+PYOORB_CONFIG = {"dynamical_model": "N", "ephemeris_file": "de430.dat"}
+
+
+class PYOORB(Backend):
+    def __init__(self, **kwargs):
+        # Make sure only the correct kwargs are passed to the constructor
+        allowed_kwargs = PYOORB_CONFIG.keys()
+        for k in kwargs:
+            if k not in allowed_kwargs:
+                warnings.warn(
+                    f"Invalid argument {k} passed to PYOORB backend. Ignoring..."
+                )
+
+        # If an allowed kwarg is missing, add the default
+        for k in allowed_kwargs:
+            if k not in kwargs:
+                kwargs[k] = PYOORB_CONFIG[k]
+
+        self.dynamical_model = kwargs["dynamical_model"]
+        self.ephemeris_file = kwargs["ephemeris_file"]
+
+        env_var = "ADAM_CORE_PYOORB"
+        if env_var in os.environ.keys() and os.environ[env_var] == "True":
+            pass
+        else:
+            if os.environ.get("OORB_DATA") is None:
+                os.environ["OORB_DATA"] = os.path.join(
+                    os.environ["CONDA_PREFIX"], "share/openorb"
```

Comment:
Please check if CONDA_PREFIX is in env vars first

Reply:
Done!